rust/bindgen.bzl

```diff
@@ -0,0 +1,156 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _rust_bindgen(ctx):
+    rust_toolchain = ctx.toolchains["@io_bazel_rules_rust//rust:toolchain"]
+
+    bindgen = ctx.executable.bindgen
+    rustfmt = rust_toolchain.rustfmt if hasattr(rust_toolchain, 'rustfmt') else None
+    clang = ctx.executable.clang
+    libclang = ctx.attr.libclang
+    libstdcxx = ctx.attr.libstdcxx
+    cc_toolchain = ctx.attr.cc_toolchain
+    cc_fragment = ctx.fragments.cpp
+
+    # nb. We can't grab the cc_library`s direct headers, so a header must be provided.
```

Comment:
This usage seems a bit complicated to me. Bindgen really just generates a header file... I feel like we shouldn't require cc_lib to use bindgen.

What motivated this change (relative to https://github.com/acmcarther/void/blob/master/tools/rust/bindgen.bzl)?

Reply:
The transitive includes can be quite sprawling when dealing with an existing codebase, and this dramatically simplifies the end-user invocation.

I agree that is is inordinately complicated, but I don't know how to better extract what we need from the cc stuff

Reply:
/cc @mhlopko is the owner of C++ on Bazel side :)

Reply:
I think this is necessary to traverse the transitive headers, I don't see a way around this.